experiments/benchmarks/benchmark_sim.jl

```diff
@@ -0,0 +1,169 @@
+# This script contains functions to assess performance, either via Nsight or by running the
+# model multiple times withj the integrated profiler and collecting statistics
+
+# You can choose between the profiling type
+# by requestion --profiler nsight or --profiler integrated. If not provided,
+# the integrated profiler is used. If benchmarking on cpu with the integrated profiler,
+# flamegraphs are created and saved. If benchmarking on gpu with the integrated profiler, the results
+# are saved to csv files. If nsight is used, the simulation is profiled for two steps.
+
+
+
+import CUDA
+using ArgParse
+using Test
+using Statistics
+import Profile, ProfileCanvas
+using Dates
+import ClimaLand
+using ClimaComms
+using CSV
+
+function parse_commandline()
+    s = ArgParseSettings()
+    @add_arg_table s begin
+        "--profiler"
+        help = "Profiler option: nsight or flamegraph"
+        arg_type = String
+        default = "integrated"
+    end
+    return parse_args(s)
+end
+
+function run_benchmarks(
+    device,
+    setup_simulation,
+    profiler,
+    PREVIOUS_GPU_TIME,
+    outdir,
+)
+    # run once just to ensure everything is compiled
+    simulation = setup_simulation()
+    ClimaLand.Simulations.solve!(simulation)
+    if profiler == "integrated"
+        @info "Running timing benchmarks"
+        (average_timing_s, std_timing_s) =
+            run_timing_benchmarks(device, setup_simulation)
+        @info "Done with timing benchmarks. Starting integrated profiling"
+        run_integrated_profiler(device, setup_simulation, outdir)
+        if get(ENV, "BUILDKITE_PIPELINE_SLUG", nothing) ==
+           "climaland-benchmark" && device isa ClimaComms.CUDADevice
+            if average_timing_s > PREVIOUS_GPU_TIME + std_timing_s
+                @info "Possible performance regression, previous average time was $(PREVIOUS_GPU_TIME)"
+            elseif average_timing_s < PREVIOUS_GPU_TIME - std_timing_s
+                @info "Possible significant performance improvement, please update PREVIOUS_GPU_TIME in $(@__DIR__)"
+            end
+            @testset "Performance" begin
```

Comment:
this makes the timing a strict check, right? 

is this now something we want to run in order to merge into main?